sw/device/freestanding/stdint.h

```diff
@@ -0,0 +1,121 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_FREESTANDING_STDINT_H_
+#define SW_DEVICE_FREESTANDING_STDINT_H_
+
+/**
+ * This header implements the stdint.h standard header, as required by C11 S4.6.
+ * This header is specified in detail in S7.20 the same.
+ *
+ * Actual limit values below are defined as:
+ * - Unsigned MAXs are defined as the |0xff..ffU| all-ones representation.
+ * - Signed MINs are defined as |0x80..00| all-zeroes-but-the-last
+ *   representation.
+ * - Unsigned MAXs are defined as |0x7f..ff| all-ones-but-the-last
+ *   representation.
+ * These shorthands are correct for two's complement and the ones least likely
+ * to get a mistake in.
+ *
+ * It goes without saying that this file assumes the underlying representation
+ * is two's complement and little-endian.
+ *
+ * See https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md
+ * for the RISC-V specifics this file conforms to.
+ */
+
+typedef unsigned char uint8_t;
+typedef unsigned short uint16_t;
+typedef unsigned int uint32_t;
+typedef unsigned long long uint64_t;
+typedef signed char int8_t;
+typedef signed short int16_t;
+typedef signed int int32_t;
+typedef signed long long int64_t;
+
+typedef uint8_t uint_least8_t;
+typedef uint16_t uint_least16_t;
+typedef uint32_t uint_least32_t;
+typedef uint64_t uint_least64_t;
+typedef int8_t int_least8_t;
+typedef int16_t int_least16_t;
+typedef int32_t int_least32_t;
+typedef int64_t int_least64_t;
+
+typedef uint8_t uint_fast8_t;
+typedef uint16_t uint_fast16_t;
+typedef uint32_t uint_fast32_t;
+typedef uint64_t uint_fast64_t;
+typedef int8_t int_fast8_t;
+typedef int16_t int_fast16_t;
+typedef int32_t int_fast32_t;
+typedef int64_t int_fast64_t;
+
+typedef unsigned long intptr_t;
+typedef signed long uintptr_t;
+
+typedef uint64_t uintmax_t;
+typedef int64_t intmax_t;
+
+// clang-format off
+// Nybble scale:      0-2-4-6-8-a-c-e-
+//                    -1-3-5-7-9-b-d-f
```

Comment:
What's this here for?

Reply:
I've expanded this to describe what the point of it is. In short, it's just the best way I can think of to allow a reader to check that each literal there is correct.